lib/tasks/email_domain_actions.rake

```diff
@@ -0,0 +1,144 @@
+# -*- encoding : utf-8 -*-
+namespace :email_domain_actions do
+
+  desc "Lists top 20 most used email domains"
+  task :top20 => :environment do
+    # the limit of 20 is fairly arbitrary but I imagine this being more
+    # annoying than useful otherwise
+
+    from = ENV["START_DATE"]
+
+    sql = if from
+      "SELECT substring(email, position('@' in email)+1) AS domain, " \
+      "COUNT(id) AS count " \
+      "FROM users " \
+      "WHERE admin_level = 'none' AND created_at >= '#{from}' " \
+      "GROUP BY domain " \
+      "ORDER BY count DESC " \
+      "LIMIT 20"
+    else
+      "SELECT substring(email, position('@' in email)+1) AS domain, " \
+      "COUNT(id) AS count " \
+      "FROM users " \
+      "WHERE admin_level = 'none' " \
+      "GROUP BY domain " \
+      "ORDER BY count DESC " \
+      "LIMIT 20"
+    end
+    results = User.connection.select_all(sql)
+
+    column1_width = results.map { |x| x["domain"].length }.sort.last
+
+    p "Since #{from}..." if from
+
+    p " domain ".ljust(column1_width + 2, " ") + " | " + " count "
+    p "--------".ljust(column1_width + 2, "-") + " | " + "-------"
+
+    results.each do |result|
+      p " #{result["domain"].ljust(column1_width, " ")}  |  #{result["count"]}"
+    end
+
+  end
+
+  desc "Lists per domain stats"
+  task :stats => :environment do
+    raise "must supply a DOMAIN value" unless ENV["DOMAIN"]
+    domain = ENV["DOMAIN"]
+    from = ENV["START_DATE"]
+
+    total_users = if from
+      User.where("email like ?", "%@#{domain}").
+           where(:admin_level => 'none').
+           where("created_at >= ?", from).
+           count
+    else
+      User.where("email like ?", "%@#{domain}").
+           where(:admin_level => 'none').
+           count
+    end
+
+    spammers = if from
```

Comment:
I don't think we should use the term spammers here (in var name or output) – users could be banned for other reasons. When looking at the PR description I had thought that this was using the spam scorer.